src/litserve/server.py

```diff
@@ -483,6 +479,24 @@ async def data_streamer():
 
             return StreamingResponse(data_streamer())
 
+        if not self._specs:
+            stream = self.app.lit_api.stream
+            # In the future we might want to differentiate endpoints for streaming vs non-streaming
+            # For now we allow either one or the other
+            endpoint = "/predict"
+            methods = ["POST"]
+            self.app.add_api_route(
+                endpoint, stream_predict if stream else predict, methods=methods, dependencies=[Depends(setup_auth())]
+            )
+
+        for spec in self._specs:
+            spec: LitSpec
+            # TODO: We need to call setup in after spawning inference workers
+            # spec.setup(self)
```

Comment:
if we call `spec.setup(self)` here then `spec` object becomes unpickleable because `LitServer.app` (FastAPI) contains a lot of local methods. We need to call setup inside the inference worker (after the process starts). 

cc: @lantiga